components/apimgt/org.wso2.carbon.apimgt.rest.api.authenticator/src/main/java/org/wso2/carbon/apimgt/rest/api/authenticator/AuthenticatorService.java

```
@@ -0,0 +1,364 @@
+            throws APIManagementException {
+        JsonObject oAuthData = new JsonObject();
+        List<String> grantTypes = new ArrayList<>();
+        grantTypes.add(KeyManagerConstants.PASSWORD_GRANT_TYPE);
+        grantTypes.add(KeyManagerConstants.AUTHORIZATION_CODE_GRANT_TYPE);
+        grantTypes.add(KeyManagerConstants.REFRESH_GRANT_TYPE);
+        APIMStoreConfigurations storeConfigs = ServiceReferenceHolder.getInstance().getAPIMStoreConfiguration();
+        String callBackURL = storeConfigs.getApimBaseUrl() + "login/callback/" + appName;
+        // Get scopes of the application
+        String scopes = getApplicationScopes(appName);
+        if (log.isDebugEnabled()) {
+            log.debug("Set scopes for " + appName + " application using swagger definition.");
+        }
+        OAuthApplicationInfo oAuthApplicationInfo;
+        try {
+            oAuthApplicationInfo = createDCRApplication(appName, callBackURL, grantTypes);
+            if (oAuthApplicationInfo != null) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Created DCR Application successfully for " + appName + ".");
+                }
+                String oAuthApplicationClientId = oAuthApplicationInfo.getClientId();
+                String oAuthApplicationCallBackURL = oAuthApplicationInfo.getCallBackURL();
+                oAuthData.addProperty(KeyManagerConstants.OAUTH_CLIENT_ID, oAuthApplicationClientId);
+                oAuthData.addProperty(KeyManagerConstants.OAUTH_CALLBACK_URIS, oAuthApplicationCallBackURL);
+                oAuthData.addProperty(KeyManagerConstants.TOKEN_SCOPES, scopes);
+                oAuthData.addProperty(KeyManagerConstants.AUTHORIZATION_ENDPOINT,
+                        storeConfigs.getAuthorizationEndpoint());
+                oAuthData.addProperty("is_sso_enabled", storeConfigs.isSsoEnabled());
+            } else {
+                String errorMsg = "No information available in OAuth application.";
```

Do we need to throw a exception here?